client/containers/Pub/PubDocument/PubMouseEvents/PubMouseEvents.js

```diff
@@ -0,0 +1,122 @@
+import React, { useEffect, useRef, useReducer } from 'react';
+import PropTypes from 'prop-types';
+import { PubNoteContent } from 'components';
+import NotePopover from './NotePopover';
+import HeaderPopover from './HeaderPopover';
+
+const propTypes = {
+	pubData: PropTypes.object.isRequired,
+	collabData: PropTypes.object.isRequired,
+	locationData: PropTypes.object.isRequired,
+	historyData: PropTypes.object.isRequired,
+	mainContentRef: PropTypes.object.isRequired,
+};
+
+const PubMouseEvents = (props) => {
+	const { pubData, collabData, historyData, mainContentRef, locationData } = props;
+	const timeouts = useRef({});
+	const [hoverElems, hoverElemsDispatch] = useReducer((state, action) => {
+		return {
+			...state,
+			[action.type]: action.elem,
+		};
+	}, {});
+
+	/* Specify the types of elems we want events for */
+	const mouseElemTypes = [
```

Comment:
Break this constant out of the function body?

Reply:
Handled in d6864b3.